torchvision/csrc/io/image/cpu/exif.h

```diff
@@ -65,7 +66,22 @@ constexpr uint16_t REQ_EXIF_TAG_MARK = 0x2a;
 constexpr uint16_t ORIENTATION_EXIF_TAG = 0x0112;
 constexpr uint16_t INCORRECT_TAG = -1;
 
-inline uint16_t get_endianness(const std::vector<unsigned char>& exif_data) {
+class ExifDataReader {
+ public:
+  ExifDataReader(unsigned char* p, size_t s) : _ptr(p), _size(s) {}
+  size_t size() const {
+    return _size;
+  }
+  const unsigned char& operator[](size_t index) const {
+    return _ptr[index];
```

Comment:
Should this enforce that `0 <= index < size` ? I know there are some levels of checks in the getters like `get_uint16` already, but a memory access without a check seems a bit risky?

Reply:
We can add that. Shall we use TORCH_CHECK ?

Reply:
sure